concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+import os
+import sys
+import time
+from datetime import datetime
+from threading import Thread
+from typing import Optional
+
+import requests
+from golem_messages import message
+from golem_messages.factories.tasks import ReportComputedTaskFactory
+from golem_messages.message.concents import ForceGetTaskResult
+from golem_messages.message.concents import ForceReportComputedTask
+from golem_messages.message.tasks import AckReportComputedTask
+from golem_messages.message.tasks import ReportComputedTask
+from golem_messages.message.tasks import TaskToCompute
+
+from api_testing_common import PROVIDER_PRIVATE_KEY
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import api_request
+from api_testing_common import assert_condition
+from api_testing_common import create_signed_task_to_compute
+from api_testing_common import run_tests
+from api_testing_common import timestamp_to_isoformat
+from common.helpers import get_current_utc_timestamp
+from common.helpers import sign_message
+
+"""
+Important: to correctly run this tests and create your own tests like this please remember:
+1. TestApi should be in all class names
+2. test_that should be in all test methods in classes names
+"""
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+REPORT_COMPUTED_TASK_SIZE = 10
+
+responses = []  # type: list
+
+NUMBER_OF_TESTING_THREADS = 3  # changing this variable remember to correct number of expected responses
+MAXIMUM_WAITING_TIME_FOR_ALL_RESPONSES = 3  # seconds
+
+
+def get_unique_subtask_id_for_tests() -> str:
+    now = datetime.now()
+    return str(now.year) + str(now.month) + str(now.day) + str(now.hour) + str(now.minute) + str(now.second) + str(
+        now.microsecond)
+
+
+def call_function_in_treads(
+        func,
+        number_of_threads: int,
+        *args,
+        **kwargs,
+) -> None:
+    for i in range(number_of_threads):
+        t = Thread(target=func, args=(*args,), kwargs={**kwargs})
+        t.start()
+
+
+def get_force_report_computed_task(
+    task_to_compute: TaskToCompute,
+    report_computed_task: ReportComputedTask,
+) -> ForceReportComputedTask:
+    force_report_computed_task = ForceReportComputedTask()
+    force_report_computed_task.report_computed_task = report_computed_task
+    force_report_computed_task.report_computed_task.task_to_compute = task_to_compute
```

Comment:
There should not be task_to_compute added to others message this way. Please try to do this like in real flow:
1) Create TaskToCompute and sign it
2) Create ReportComputedTask with signed TaskToCompute
3) Create ForceReportComputedTask with signed ReportComputedTask from point 2